### DISCUSSION TITLE
Clone entity configs or create entity config from messages

### DISCUSSION THREAD

**Discussion by Alex9779** [zachowj/node-red-contrib-home-assistant-websocket#1230] [Q&A] [answered]
I recently overhauled my electric fence monitoring system and tried to make it easier to maintain.
The sensors run on a LoRaWAN with a Chirpstack backend so I am subscribing the MQTT messages on the Chirpstack server to get the sensor uplinks.
So basically the sensors are defined in Chirpstack and I want to transport them to HA while using Node-Red to manage the logic to interpret the sensor values to states like "Normal", "Low", "Critical" and also monitoring devices not sending uplinks or battery level monitoring.

I came up with a sub flow and some function nodes which implement some kind of threshold and hysteresis logic and this works fine.

But it is still a bunch of work to create new sensors because of the entity config.
Is there a way to "clone" and existing entity config and just change the names? If you copy a "sensor" node, the assingement of the entity config node is also copied to it is a 1:1 copy, then I want to change the entity config but all I can do is create a new one from scratch which is annoying if you just want to change a counter in the name and friendly name.

Another question that came to my mind is, is it possible to create entity config with input from other nodes?
That way I could futher simplify my flows, currenty I have a "flow" for every single sensor just because of that each sensor has a bound entity config and entity configs have to be created manually as far as I know...
Adding a new sensor in Chirpstack would then need no action in Node-Red and the sensor could be automatically published to HA.

**Comment by zachowj** [comment, marked as answer]
> Is there a way to "clone" and existing entity config and just change the names?

You can export the entity config and import a copy of it.

> Another question that came to my mind is, is it possible to create entity config with input from other nodes?

This can be done using an API node to send HA the expected output from an entity config. Here's an example <https://community.home-assistant.io/t/help-merging-2-node-red-flows-in-1/154642/23>

If you're already using MQTT it would probably be easier to use the MQTT sensor <https://www.home-assistant.io/integrations/sensor.mqtt/>

**Comment by Alex9779** [comment]
Thanks for the hints, though for cloning a UI button or something would be a nice addition.

Didn't think of MQTT, this is an interesting suggestion, I could use the MQTT auto discovery to create sensors on the fly. Thx!
